http/middleware.go

```diff
@@ -151,3 +152,39 @@ func (b *bodyEchoer) Read(p []byte) (int, error) {
 func (b *bodyEchoer) Close() error {
 	return b.rc.Close()
 }
+
+func getMembersPrecheckMiddleware(next http.HandlerFunc, backend MemberBackend) http.HandlerFunc {
```

Comment:
nit: the signature here you may want to consider using a closure and returning the middleware func. Its a popular pattern in a lot of http middleware in go. Check out the following: https://github.com/influxdata/influxdb/blob/master/http/middleware.go#L15

the func closes over the logger here, and returns a middleware (func(http.handler) http.Handler), which allows you to chain these things very easily. In your case it would look like:

```go
func getMembersPrecheckMiddleware(backend MemberBackend) func(http.Handler) http.Handler {
   return func(next http.Handler) http.Handler {
      fn := func(w http.ResponseWriter, r *http.Request) {
         // req handler stuffs yada yada
      }

     return http.HandlerFunc(fn)
   }
```

in hindsight, I wish I would have made the following type to illuminate that better:

```go
type Middleware func(next http.handler) http.Handler
```